lib/dialects/postgres/query-generator.js

```diff
@@ -436,18 +436,32 @@ var QueryGenerator = {
   },
 
   showIndexesQuery: function(tableName) {
+    var query,schemaName;
     if (!Utils._.isString(tableName)) {
+      schemaName = tableName.schema;
       tableName = tableName.tableName;
     }
 
-    // This is ARCANE!
-    var query = 'SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, ' +
-      'array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) ' +
-      'AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a ' +
-      'WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND '+
-      "t.relkind = 'r' and t.relname = '<%= tableName %>' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;";
+    if(schemaName){
+      query = 'SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, ' +
+        'array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) ' +
+        'AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a, pg_namespace s ' +
+        'WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND ' +
+        "t.relkind = 'r' and t.relname = '<%= tableName %>' AND s.oid = t.relnamespace AND s.nspname = '<%= schemaName %>' " +
```

Comment:
Not very DRY ;) 

It would be better to separate the schema part into a separate part and concat that into the part of the query thats shared between both